src/layout/dom.ts

```diff
@@ -156,6 +185,34 @@ export function update(node: Node, data: NodeInfo, source: Source): void {
     }
 }
 
+function mask(data: NodeInfo, masked: boolean): boolean {
+    let attributes = data.attributes;
+    let tag = data.tag.toUpperCase();
+
+    // Do not proceed if attributes are missing for the node
+    if (attributes === null || attributes === undefined) { return masked; }
+
+    // Check for blacklist fields (e.g. address, phone, etc.) only if the input node is not already masked
+    if (masked === false && tag === Constant.TAG_INPUT && Constant.NAME_ATTRIBUTE in attributes) {
+        let value = attributes[Constant.NAME_ATTRIBUTE].toLowerCase();
+        for (let name of BLACKLIST_NAMES) {
```

Comment:
Should we do this using CSS selectors instead? Would give us the option of configuring masking by CSS Selectors as a general feature for later like other system provide. 
This is an interesting model: [https://help.fullstory.com/hc/en-us/articles/360044349073-FullStory-Private-by-Default](https://help.fullstory.com/hc/en-us/articles/360044349073-FullStory-Private-by-Default)

Reply:
Will do so in subsequent release.

Reply:
Simple name change. Can you just does this now before you check in?

---
In reply to: [405123883](https://github.com/microsoft/clarity-js/pull/172#discussion_r405123883) [](ancestors = 405123883)